### PR TITLE
Adding Interface for mock testing

### PR DESCRIPTION
**Pull request by dciborow**
## Description

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [ ] Pull request includes test coverage for the included changes.
- [ ] The code builds without any errors.

### CODE REVIEW

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
  */
 package com.microsoft.azure.eventhubs;
 
+import com.microsoft.azure.servicebus.*;
```
Comment (SreeramGarlapati) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java]: please leave these expanded and after the java class library namespaces.. :)

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/IEventHubClient.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
+package com.microsoft.azure.eventhubs;
```
Comment (SreeramGarlapati) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/IEventHubClient.java]: @dciborow@microsoft.com : add mit license header (copy from other files...)

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/IEventHubClient.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+import com.microsoft.azure.servicebus.ServiceBusException;
+
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Interface for Event Hub Client
+ *
+ * Created by dcibo on 6/6/2017.
```
Comment (SreeramGarlapati) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/IEventHubClient.java]: Please remove this..

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/IEventHubClient.java (hunk 2 above)
Comment (dciborow) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/IEventHubClient.java]: fixed in 2nd commit

**Comment by SreeramGarlapati** [comment]
:shipit:

**Pull request merged**